test/gtest/ucm/malloc_hook.cc

```diff
@@ -771,7 +771,17 @@ class mmap_hooks {
 
         /* 8. sbrk call - single thread only */
         {
-            if (!RUNNING_ON_VALGRIND && m_num_threads < 2) {
+            int num_threads = 0;
+            ucs::enum_threads(enum_threads_cb, &num_threads);
+            // check for real number of threads:
+            // test is actual only in case if 3 threads are in process:
```

Comment:
use sbrk() only if there are 3 threads in the system: ....
otherwise, another thread can call use malloc/free in same time, leading to heap corruption.

Reply:
comment updated